core/iwasm/common/wasm_runtime_common.c

```diff
@@ -1267,6 +1267,9 @@ wasm_runtime_is_built_in_module(const char *module_name)
             || !strcmp("wasi_snapshot_preview1", module_name)
 #if WASM_ENABLE_SPEC_TEST != 0
             || !strcmp("spectest", module_name)
+#endif
+#if WASM_ENABLE_LIB_WASI_THREADS != 0
+            || !strcmp("foo", module_name)
```

Comment:
wondering why explicit "foo" here? If it came from a test case, better be surrounded by a `#if...#endif`.

Reply:
Yes, maybe we can add a cmake variable `cmake -DWAMR_BUILD_WASI_TEST=1/0` and add gcc macro `WASM_ENABLE_WASI_TEST`, like WAMR_BUILD_SPEC_TEST. And enable it when testing wasi_certification in test_wamr.sh (just a guess, to be confirmed):
```diff
diff --git a/tests/wamr-test-suites/test_wamr.sh b/tests/wamr-test-suites/test_wamr.sh
index 135940e1..bbfd5259 100755
--- a/tests/wamr-test-suites/test_wamr.sh
+++ b/tests/wamr-test-suites/test_wamr.sh
@@ -1003,6 +1003,9 @@ function trigger()
                 EXTRA_COMPILE_FLAGS+=" -DWAMR_BUILD_LIBC_UVWASI=0 -DWAMR_BUILD_LIBC_WASI=1"
                 break
             fi
+            if [[ "$test" == "wasi_certification" ]]
+                EXTRA_COMPILE_FLAGS+=" -DWAMR_BUILD_WASI_TEST=1"
+            fi
```

Reply:
I have added the cmake variable `WAMR_BUILD_WASI_TEST` and the corresponding GCC macro `WASM_ENABLE_WASI_TEST`. Since I'm not very familiar with this part, please let me know if there's anything missing or if further adjustments are needed.

And I have no idea about the failing tests. Could you please help me take a look?

Reply:
Add lines in core/config.h:
```C
#ifndef WASM_ENABLE_WASI_TEST
#define WASM_ENABLE_WASI_TEST 0
#endif
```